### PR TITLE
feat: add Ranking Question to the API

### LINKED ISSUE #3230: [FEATURE] API Ranking Question
**Is your feature request related to a problem? Please describe.**

Ranking Questions cannot be created in the API for a Feedback Dataset.

**Describe the solution you'd like**

Update the API so Ranking Questions can be created in the API for Feedback Datasets.

### PR DESCRIPTION
**Pull request by gabrielmbmb**
# Description

This PR adds the required classes to create a Ranking Question and to validate their responses in the API. Additionally, I've refactor the `LabelSelectionQuestionSettingsOption` class to be named as `ValueTextQuestionSettingsOption`, as this same class can be used in the Ranking Question.

Closes #3230

**Type of change**

- [x] New feature (non-breaking change which adds functionality)
- [x] Refactor

**How Has This Been Tested**

Unit tests have been added covering the creation of the Ranking Question as well as the validation of the submitted responses for this kind of question.

**Checklist**

- [x] I have merged the original branch into my forked branch
- [ ] I added relevant documentation
- [x] follows the style guidelines of this project
- [x] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

### CODE REVIEW

File: src/argilla/server/search_engine.py
Comment (gabrielmbmb) [inline src/argilla/server/search_engine.py:259]: not sure if this is the best mapping type that we can use for the Ranking Question @frascuchon

File: CHANGELOG.md
Comment (frascuchon) [inline CHANGELOG.md]: Maybe make a direct reference to the API endpoint  `/api/v1/datasatet/:dataset_id/questions` now support RankingQUestionSettings, bla blablab

File: src/argilla/server/search_engine.py
Comment (frascuchon) [reply src/argilla/server/search_engine.py:259]: It makes sense in terms we can query records with a response having the option o in ranking 1 for example. Let's keep it as is.

**Review by frascuchon — APPROVED** [review]
Let's wait for the release branch before merging just to not disturb the release workflow

**Pull request merged**
